perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/GroupsManagerMethod.java

```diff
@@ -1059,13 +1063,14 @@ public Void call(ApiCaller ac, Deserializer parms) throws PerunException {
 
 		@Override
 		public List<User> call(ApiCaller ac, Deserializer parms) throws PerunException {
-			if(parms.contains("onlyDirectAdmins")) {
-				return ac.getGroupsManager().getAdmins(ac.getSession(),
-						ac.getGroupById(parms.readInt("group")),
-						parms.readBoolean("onlyDirectAdmins"));
-			} else {
-				return ac.getGroupsManager().getAdmins(ac.getSession(),
-						ac.getGroupById(parms.readInt("group")));
+			try {
+				if(parms.contains("onlyDirectAdmins")) {
+					return AuthzResolver.getAdmins(ac.getSession(), ac.getGroupById(parms.readInt("group")), Role.GROUPADMIN, parms.readBoolean("onlyDirectAdmins"));
+				} else {
+					return AuthzResolver.getAdmins(ac.getSession(), ac.getGroupById(parms.readInt("group")), Role.GROUPADMIN, false);
+				}
+			} catch (RoleCannotBeManagedException ex) {
+				throw new InternalErrorException(ex);
```

Comment:
Why can't we just throw `RoleCannotBeManagedException`?

Reply:
Old methods weren't throwing this exception, and under normal circumstances, this should not happen now either, right?

Reply:
In this case, it does not matter which method RPC throws since it didn't use to throw either of them; it just seemed redundant, but I see your point.